Source/Core/Core/PowerPC/JitCommon/JitCache.cpp

```diff
@@ -366,15 +348,15 @@ void JitBaseBlockCache::DestroyBlock(int block_num, bool invalidate)
 
 void JitBaseBlockCache::MoveBlockIntoFastCache(u32 addr, u32 msr)
 {
-  int block_num = GetBlockNumberFromStartAddress(addr, msr);
-  if (block_num < 0)
+  JitBlock* block = GetBlockFromStartAddress(addr, msr);
+  if (!block)
   {
     Jit(addr);
   }
   else
   {
-    FastLookupEntryForAddress(addr) = block_num;
-    LinkBlock(block_num);
+    FastLookupEntryForAddress(addr) = block - &blocks[0];
```

Comment:
Pointer subtraction returns a 64-bit signed type (ptrdiff_t), so this would implicitly truncate.